### PR TITLE
Add static code analysis to ci pipeline

### PR DESCRIPTION
**Pull request by Renny4Real**
Adds the [FxCop NuGet package](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers) and tags to the project's `.csproj` file to cause the production build to fail if there are any errors we want to surface from FxCop.

It currently passes as we haven't defined which errors to surface and all the warnings from FxCop are currently being suppressed.

### CODE REVIEW

**Comment by Renny4Real** [comment]
@elcorbs Yeah, it does.

It's possible that using FxCop could overlap with linting, but I'm not entirely sure. 
I haven't investigated deep into the range of categories or rules it's using to analyse the code.

We could explore this later and see if we need to remove the linting step in CI, but I'm definitely up for trimming down code where necessary.

File: README.md [hunk 1]
```diff
@@ -77,6 +77,17 @@ Then we have an automated six step deployment process, which runs in CircleCI.
...
 
+## Static Code Analysis
+
+### Using [FxCop Analysers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)
+
+FxCop runs code analysis when the Solution is built.
+
+Both the API and Test projects have been set up to **treat all warnings from the code analysis as errors** and therefore, fail the build.
+
+However, we can select which errors to suppress by setting the severity of the responsible rule to none, e.g `dotnet_analyzer_diagnostic.<Category-or-RuleId>.severity = none`, within the `.editorConfig` file.
```
Comment (elcorbs) [inline README.md:88]:
```suggestion
However, we can select which errors to suppress by setting the severity of the responsible rule to none, e.g `dotnet_analyzer_diagnostic.<Category-or-RuleId>.severity = none`, within the `.editorconfig` file.
```

File: README.md [hunk 2]
```diff
@@ -77,6 +77,17 @@ Then we have an automated six step deployment process, which runs in CircleCI.
 
 Our staging and production environments are hosted by AWS. We would deploy to production per each feature/config merged into  `master`  branch.
 
+## Static Code Analysis
```
Comment (louiseswift) [inline README.md:80]: This is a good README section: concise but clear and informative 👍

Comment (Renny4Real) [reply README.md:80]: Thanks! 😁

**Pull request merged**